src/main/java/com/google/error_analyzer/backend/StackTrace.java

```diff
@@ -0,0 +1,144 @@
+    String fileName) throws IOException {
+        SearchRequest searchRequest = createSearchRequest(fileName,
+        errorLogLineNumber);
+        logger.info("Finding stack for ".concat(errorLogLineNumber.toString()));
+        ImmutableList < SearchHit > rangeHits = logDao.getHitsFromIndex(searchRequest);
+        Builder < String > stackLogLinesBuilder = ImmutableList.< String > builder();
+        Integer startOfStack = findStartOfStack(rangeHits);
+        if (startOfStack == -1) {
+            return stackLogLinesBuilder.build();
+        } else {
+            //add message brfore stack starts to stackLogLines
+            logger.info("Adding ".concat(startOfStack.toString())
+            .concat(" messages to stack list"));
+            stackLogLinesBuilder.addAll(extractLogTextFormHits(rangeHits, 0, startOfStack));
+        }
+        ImmutableList < String > stackList = iterateHitsForFindingStack(rangeHits, startOfStack);
+        stackLogLinesBuilder.addAll(stackList);
+        if (stackList.size() < rangeHits.size() - startOfStack) {
+            return stackLogLinesBuilder.build();
+        }
+        while (true) {
+            logger.info("Exceeding batch size request");
+            Integer nextLine = errorLogLineNumber + stackLogLinesBuilder.build().size();
+            searchRequest = createSearchRequest(fileName, nextLine);
+            rangeHits = logDao.getHitsFromIndex(searchRequest);
+            stackList = iterateHitsForFindingStack(rangeHits, 0);
+            stackLogLinesBuilder.addAll(stackList);
+            if (stackList.size() < rangeHits.size() || rangeHits.size() == 0) {
+                return stackLogLinesBuilder.build();
+            }
+        }
+    }
+
+    // Iterate over range query to find call stack. Return if end is found
+    private ImmutableList < String > iterateHitsForFindingStack (ImmutableList < SearchHit > hits,
+    Integer start) {
+        Builder < String > stackLogLinesBuilder = ImmutableList.< String > builder();
+        for (int i = start; i < hits.size(); i++) {
+            Map < String ,Object > sourceMap = hits.get(i).getSourceAsMap();
+            String logText = (String) sourceMap.get(LogFields.LOG_TEXT);
+            Integer logLineNumber = (Integer) sourceMap.get(LogFields.LOG_LINE_NUMBER);
+            if (StackTraceFormat.matchesFormat(logText)) {
+                logger.info("Adding line:"
+                .concat(logLineNumber.toString()).concat(" to stack list"));
+                stackLogLinesBuilder.add(logText);
+            } else {
+                logger.info("End of stack trace found");
+                return stackLogLinesBuilder.build();
+            }
+        }
+        return stackLogLinesBuilder.build();
+    }
+
+    //returns index of the first searchHit that matches the stack format
+    // -1 if start is not found
+    private Integer findStartOfStack(ImmutableList < SearchHit > hits) {
+        Integer end = Math.min(ALLOWED_MESSAGES+1, hits.size());
```

Comment:
Should it be ALLOWED_MESSAGES or (ALLOWED_MESSAGES+1)?

Reply:
If we are allowing 5 messages before call stack, then at least 6 lines need to be checked